GoogleForCreators/web-stories-wp
# Crop leading & trailing whitespace

## Summary



## Relevant Technical Choices



## To-do



## User-facing changes

Text elements line-height works now more like a line-gap, there is no whitespace at the top of the first line and bottom of the last line.

## Testing Instructions

Add text with 2+ lines of text and set line-height to 2+ (5+ to really see the difference with previous behavior).

Testing old stories:

How should it work?
> Let's use the term beta-text for text added/modified using the beta plugin and stable-text for text modified with stable plugin and after (new cropping).
for beta-text that the user never touches after upgrading to stable -- nothing should change
for beta-text that the user modifies the text only, not the formatting -- nothing should change
for beta-text that the user changes styling (font family, font size, font weight, etc.) -- we recalculate cropping/etc using the new formula and in essence turn it into beta-text using their newly set formatting preferences.
for newly added beta-text, just use the new framework.

1) Create a story on the main branch (beta2) then switch to #4247 (with cropping) and play around.
2) Open the old story on #4247 and play around.

---



Fixes #2788
Fixes #3530
